src/amo/components/LandingPage.js

```diff
@@ -41,9 +44,33 @@ export class LandingPageBase extends React.Component {
 
   static defaultProps = {
     apiAddonType: getApiAddonType,
+    contentForType: null,
   }
 
-  contentForType(visibleAddonType) {
+  componentWillMount() {
+    this.setViewContextType();
+  }
+
+  componentDidUpdate() {
+    this.setViewContextType();
+  }
+
+  setViewContextType() {
+    const { apiAddonType, dispatch, params } = this.props;
+
+    try {
+      const addonType = apiAddonType(params.visibleAddonType);
+      dispatch(setViewContext(addonType));
+    } catch (err) {
+      if (err instanceof AddonTypeNotFound) {
+        log.info('AddonTypeNotFound in setViewContextType()', err);
```

Comment:
I don't think so. I'll double-check but throwing the error would cause something similar to what happened with #2545; if a component throws and we don't catch it the page will 500.